src/EnergyPlus/CurveManager.hh

```diff
@@ -453,36 +452,29 @@ namespace Curve {
 
 struct CurveManagerData : BaseGlobalStruct
 {
-    int NumCurves = 0;
-    bool GetCurvesInputFlag = true;
     bool CurveValueMyBeginTimeStepFlag = false;
     bool FrictionFactorErrorHasOccurred = false;
     bool showFallbackMessage = true;
-    EPVector<Curve::Curve *> PerfCurve;
+    Array1D<Curve::Curve *> curves;
+    std::map<std::string, int> curveMap;
+  
     Curve::BtwxtManager btwxtManager;
-    std::unordered_map<std::string, std::string> UniqueCurveNames;
-
-    void allocateCurveVector(int const count)
-    {
-        this->NumCurves = count;
-        for (int curveIndex = 1; curveIndex <= count; curveIndex++)
-            this->PerfCurve.push_back(new EnergyPlus::Curve::Curve);
-    }
 
     void init_constant_state([[maybe_unused]] EnergyPlusData &state) override
     {
     }
 
-    void init_state([[maybe_unused]] EnergyPlusData &state) override
+    void init_state(EnergyPlusData &state) override
```

Comment:
`GetCurveInput` is now part of `init_state` so it doesn't have to be called from any other function.